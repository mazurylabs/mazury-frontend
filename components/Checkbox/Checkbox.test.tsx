import { fireEvent, render, screen } from '@testing-library/react';
import { useState } from 'react';
import { Checkbox } from '.';

const WrappedCheckbox = () => {
  const [checked, setChecked] = useState(false);

  return (
    <Checkbox
      checked={checked}
      setChecked={setChecked}
      label="Label"
      id="checkbox-1"
    />
  );
};

describe('Checkbox', () => {
  test('renders and behaves correctly', () => {
    render(<WrappedCheckbox />);
    const outerContainer = screen.getByTestId('checkbox-outer-container');
    expect(outerContainer).toMatchSnapshot();
    const hiddenCheckbox = screen.getByTestId(
      'Hidden Checkbox'
    ) as HTMLInputElement;
    expect(hiddenCheckbox.checked).toBe(false);
    expect(hiddenCheckbox).toBeInTheDocument();
  });
});
