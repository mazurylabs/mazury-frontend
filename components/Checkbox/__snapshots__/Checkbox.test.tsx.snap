// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Checkbox renders and behaves correctly 1`] = `
<div
  class="flex items-center space-x-[11px] undefined"
  data-testid="checkbox-outer-container"
>
  <button
    class="h-6 h-4 hover:cursor-pointer"
    type="button"
  />
  <label
    class="text-base font-medium text-indigoGray-70"
    for="checkbox-1"
  >
    Label
  </label>
  <input
    data-testid="Hidden Checkbox"
    hidden=""
    id="checkbox-1"
    type="checkbox"
  />
</div>
`;
